lang/TUTORIAL_FRENCH.md

```diff
@@ -0,0 +1,676 @@
+- Règle n° 10 : utilisez des types scalaires personnalisés lorsque vous exposez quelque chose ayant une valeur sémantique spécifique.
+- Règle n° 11 : Utilisez des enums pour les champs qui ne peuvent prendre qu'un ensemble spécifique de valeurs.
+- Règle n° 12 : l'API doit fournir une logique d'entreprise, et pas seulement des données. Les calculs complexes doivent être effectués sur le serveur, à un seul endroit, et non sur le client, à plusieurs endroits.
+- Règle n° 13 : Fournissez également les données brutes, même si elles sont accompagnées d'une logique métier.
+- Règle n° 14 : écrivez des mutations distinctes pour des actions logiques distinctes sur une ressource.
+- Règle n° 15 : La mutation des relations est vraiment compliquée et ne se résume pas facilement à une règle rapide.
+- Règle n° 16 : lorsque vous écrivez des mutations distinctes pour des relations, demandez-vous s'il serait utile que les mutations opèrent sur plusieurs éléments à la fois.
+- Règle n° 17 : Préfixez les noms de mutations par l'objet qu'elles mutent pour un regroupement alphabétique (par exemple, utilisez `orderCancel` au lieu de `cancelOrder`).
+- Règle n° 18 : Ne rendez les champs d'entrée obligatoires que s'ils sont réellement nécessaires, d'un point de vue sémantique, à la poursuite de la mutation.
+- Règle n°19 : Utilisez des types plus faibles pour les entrées (par exemple String au lieu de Email) lorsque le format est sans ambiguïté et que la validation côté client est complexe. Cela permet au serveur d'exécuter toutes les validations non triviales en une seule fois et de renvoyer les erreurs à un seul endroit et dans un seul format, ce qui simplifie le client.
+- Règle n° 20 : utilisez des types plus forts pour les entrées (par exemple, DateTime au lieu de String) lorsque le format peut être ambigu et que la validation côté client est simple. Cela apporte de la clarté et encourage les clients à utiliser des contrôles de saisie plus stricts (par exemple, un widget de sélection de date au lieu d'un champ de texte libre).
+- Règle n° 21 : Structurez les entrées de mutation de manière à réduire les doublons, même si cela exige de relâcher les contraintes de rigueur sur certains champs.
+- Règle n° 22 : les mutations doivent fournir des erreurs au niveau de l'utilisateur ou de l'entreprise via un champ userErrors dans les données utiles de la mutation. L'entrée d'erreurs de requête de niveau supérieur est réservée aux erreurs de niveau client et serveur.
+- Règle n° 23 : La plupart des champs de données utiles d'une mutation doivent être annulables, à moins qu'il n'y ait vraiment une valeur à renvoyer dans tous les cas d'erreur possibles.
+
+## Conclusion
+
+Merci d'avoir lu notre tutoriel ! Nous espérons qu'à ce stade, vous avez une idée solide de la façon de concevoir une bonne API GraphQL.
+
+Une fois que vous avez conçu une API dont vous êtes satisfait, il est temps de la mettre en œuvre !
```

Comment:
I would suggest mass replacing:
- `""` with `«»` (proper French quotes)
- `'` with `’` (curly quotes, looks much nicer in most fonts)